src/model/CoilCoolingDXCurveFitOperatingMode.cpp

```diff
@@ -0,0 +1,545 @@
+  }
+
+  void CoilCoolingDXCurveFitOperatingMode_Impl::autosizeNominalEvaporativeCondenserPumpPower() {
+    bool result = setString(OS_Coil_Cooling_DX_CurveFit_OperatingModeFields::NominalEvaporativeCondenserPumpPower, "autosize");
+    OS_ASSERT(result);
+  }
+
+  bool CoilCoolingDXCurveFitOperatingMode_Impl::setNominalSpeedNumber(int nominalSpeedNumber) {
+    bool result = setInt(OS_Coil_Cooling_DX_CurveFit_OperatingModeFields::NominalSpeedNumber, nominalSpeedNumber);
+    return result;
+  }
+
+  void CoilCoolingDXCurveFitOperatingMode_Impl::resetNominalSpeedNumber() {
+    bool result = setString(OS_Coil_Cooling_DX_CurveFit_OperatingModeFields::NominalSpeedNumber, "");
+    OS_ASSERT(result);
+  }
+
+  void CoilCoolingDXCurveFitOperatingMode_Impl::addSpeed(CoilCoolingDXCurveFitSpeed& speed) {
+    auto group = getObject<ModelObject>().pushExtensibleGroup().cast<WorkspaceExtensibleGroup>();
+    OS_ASSERT(!group.empty());
+    group.setPointer(OS_Coil_Cooling_DX_CurveFit_OperatingModeExtensibleFields::Speed,speed.handle());
+  }
+
+  boost::optional<double> CoilCoolingDXCurveFitOperatingMode_Impl::autosizedRatedGrossTotalCoolingCapacity() {
+    return getAutosizedValue("TODO_CHECK_SQL Rated Gross Total Cooling Capacity", "W");
+  }
+
+  boost::optional<double> CoilCoolingDXCurveFitOperatingMode_Impl::autosizedRatedEvaporatorAirFlowRate() {
+    return getAutosizedValue("TODO_CHECK_SQL Rated Evaporator Air Flow Rate", "m3/s");
+  }
+
+  boost::optional<double> CoilCoolingDXCurveFitOperatingMode_Impl::autosizedRatedCondenserAirFlowRate() {
+    return getAutosizedValue("TODO_CHECK_SQL Rated Condenser Air Flow Rate", "m3/s");
+  }
+
+  boost::optional<double> CoilCoolingDXCurveFitOperatingMode_Impl::autosizedNominalEvaporativeCondenserPumpPower() {
+    return getAutosizedValue("TODO_CHECK_SQL Nominal Evaporative Condenser Pump Power", "W");
+  }
+
+  void CoilCoolingDXCurveFitOperatingMode_Impl::autosize() {
+    autosizedRatedGrossTotalCoolingCapacity();
+    autosizedRatedEvaporatorAirFlowRate();
+    autosizedRatedCondenserAirFlowRate();
+    autosizedNominalEvaporativeCondenserPumpPower();
+  }
+
+  void CoilCoolingDXCurveFitOperatingMode_Impl::applySizingValues() {
+    boost::optional<double> val;
+    val = autosizedRatedGrossTotalCoolingCapacity();
+    if (val) {
+      setRatedGrossTotalCoolingCapacity(val.get());
+    }
+
+    val = autosizedRatedEvaporatorAirFlowRate();
+    if (val) {
+      setRatedEvaporatorAirFlowRate(val.get());
+    }
+
+    val = autosizedRatedCondenserAirFlowRate();
+    if (val) {
+      setRatedCondenserAirFlowRate(val.get());
+    }
+
+    val = autosizedNominalEvaporativeCondenserPumpPower();
+    if (val) {
+      setNominalEvaporativeCondenserPumpPower(val.get());
+    }
+
+  }
+
+  ModelObject CoilCoolingDXCurveFitOperatingMode_Impl::clone(Model model) const {
+    auto t_clone = ModelObject_Impl::clone(model).cast<CoilCoolingDXCurveFitOperatingMode>();
+
+    auto t_speeds = speeds();
```

Comment:
I'm not 100% sure this is necessary. Technically I think the speeds being listed as Children and being Resource objects should be handled automatically no?